annotated.go

```diff
@@ -1265,8 +1279,11 @@ func (at *asAnnotation) results(ann *annotated) (
 			continue
 		}
 
-		if !t.Implements(at.types[i].typ) {
-			return nil, nil, fmt.Errorf("invalid fx.As: %v does not implement %v", t, at.types[i])
+		if !((at.types[i].typ.Kind() == reflect.Interface && t.Implements(at.types[i].typ)) ||
+			t.ConvertibleTo(at.types[i].typ)) {
+			return nil,
+				nil,
+				fmt.Errorf("invalid fx.As: %v does not implement or is not convertible to %v", t, at.types[i])
```

Comment:
Should we make these two separate checks? This will keep the code cleaner and give better error messages.

```
if at.types[i].typ.Kind() == reflect.Interface {
    if !t.Implements(at.types[i].typ) {
        return nil, nil, fmt.Errorf("invalid fx.As: %v does not implement %v", t, at.types[i])
    }
} else if !t.ConvertibleTo(at.types[i].typ) {
        return nil, nil, fmt.Errorf("invalid fx.As: %v cannot be converted to %v", t, at.types[i])
}
```